zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/user/DbUserState.java

```diff
@@ -72,4 +72,13 @@ public UserRecord getUser(final DirectBuffer username) {
   public UserRecord getUser(final String username) {
     return getUser(wrapString(username));
   }
+
+  @Override
+  public void updateUser(final UserRecord user) {
+    username.wrapBuffer(user.getUsernameBuffer());
+    final var key = userKeyByUsernameColumnFamily.get(username);
+    persistedUser.setUser(user);
```

Comment:
Here I've set the complete user object, should this be handled in a different way @remcowesterhoud if we don't want a field updated i.e. `username`? I've seen in other places that we could potentially add another method in like `.setUserWithoutUsername` but I'm curious to hear your thoughts

Reply:
I think it's fine in this case. We retrieve the persistedUser using the username. so there's no way that it could be different and accidentally updated to a different username.